### PR TITLE
Update License to EPL-2.0 and fixing build

### PR DESCRIPTION
**Pull request by sunix**
- EPL 2.0: https://github.com/eclipse/che/issues/10465
- Fixing build: https://github.com/eclipse/che/issues/10810
- adding tslint config and apply suggestions

### CODE REVIEW

File: browser-app/package.json
Comment (evidolob) [inline browser-app/package.json]: I prefer to have fixed version on  Theia dependencies

File: che-theia-factory/package.json
Comment (benoitf) [inline che-theia-factory/package.json]: we can't use `^` as we need CQ for each dependency

File: browser-app/package.json
Comment (sunix) [reply browser-app/package.json]: Please give your reasons.
I don't think there is any problem in using latest 0.3.x version and we will have feedback from CI sooner, get errors sooner and fix them sooner. Also the way we add the extensions to the Theia image won't use this version of theia.

File: che-theia-factory/package.json
Comment (sunix) [reply che-theia-factory/package.json]: OK thanks ... but how do we do with all the dependencies that comes with Theia ?

Comment (skabashnyuk) [reply che-theia-factory/package.json]: @evidolob I think you told me that this is devDependencies and we have v+above CQ for it? Am I right?

Comment (evidolob) [reply che-theia-factory/package.json]: Unfortunately we have CQ only for [rimraf 2.6.2](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15803), for TypeScript CQ still not [approved ](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16945)

Comment (sunix) [reply che-theia-factory/package.json]: I can remove the carets  ^. Will it be fine to merge then ? or do we need to wait for CQ approval ?

Comment (benoitf) [reply che-theia-factory/package.json]: I will ping PMC for TypeScript, should be quick to approve

File: browser-app/package.json
Comment (benoitf) [reply browser-app/package.json]: AFAIK it's mainly due to CQs (waiting Theia becoming Eclipse project)

**Comment by sunix** [comment]
@benoitf @evidolob I have updated the PR: use strict version of typescript and theia dependencies

**Comment by sunix** [comment]
@benoitf @evidolob CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16945 has been approved

**Comment by benoitf** [comment]
yes I ping Sharon yesterday for CQ

**Review by benoitf — APPROVED** [review]

**Pull request merged**
